packages/frontend/tests/unit/service/channels/utils/-vote-sorter-test.ts

```diff
@@ -0,0 +1,161 @@
+      );
+
+      assert.ok(keyEquals(result[VOTE_ORDERING.targetKey], currentUser.publicKey));
+      assert.equal(result[VOTE_ORDERING.action], VOTE_ACTION.ADD);
+      assert.ok(keyEquals(result[VOTE_ORDERING.voterSigningKey], currentUser.publicSigningKey));
+      assert.ok(keyEquals(result[VOTE_ORDERING.previousChainSignature], firstVote.signature));
+      assert.equal(Object.keys(result).length, 7);
+    });
+
+    test('when ran without previous vote', function (assert) {
+      const store = getStore();
+      let yes1 = store.createRecord('identity', {
+        publicKey: new Uint8Array([21, 32]),
+      });
+      let yes2 = store.createRecord('identity', {
+        publicKey: new Uint8Array([11, 7]),
+      });
+      let no1 = store.createRecord('identity', {
+        publicKey: new Uint8Array([47, 75]),
+      });
+      let no2 = store.createRecord('identity', {
+        publicKey: new Uint8Array([32, 19]),
+      });
+      let remaining1 = store.createRecord('identity', {
+        publicKey: new Uint8Array([54, 32]),
+      });
+      let remaining2 = store.createRecord('identity', {
+        publicKey: new Uint8Array([20, 98]),
+      });
+      let currentUser = store.createRecord('identity', {
+        publicKey: convertObjectToUint8Array('key'),
+        publicSigningKey: convertObjectToUint8Array('signingKey'),
+      });
+      let currentVote = store.createRecord('vote-chain', {
+        action: VOTE_ACTION.ADD,
+        key: currentUser,
+        target: currentUser,
+        yes: [yes1, yes2],
+        no: [no1, no2],
+        remaining: [remaining2, remaining1],
+      });
+
+      let result = convertUint8ArrayToObject(generateSortedVote(currentVote));
+      assert.ok(
+        result[VOTE_ORDERING.remaining].every(
+          (current: Uint8Array, index: number, array: Uint8Array[]) =>
+            !index || array[index - 1] <= current
+        ),
+        'ensure remaining keys are sorted'
+      );
+      assert.ok(
+        result[VOTE_ORDERING.yes].every(
+          (current: Uint8Array, index: number, array: Uint8Array[]) =>
+            !index || array[index - 1] <= current
+        ),
+        'ensure yes keys are sorted'
+      );
+      assert.ok(
+        result[VOTE_ORDERING.no].every(
+          (current: Uint8Array, index: number, array: Uint8Array[]) =>
+            !index || array[index - 1] <= current
+        ),
+        'ensure no keys are sorted'
+      );
+
+      assert.ok(keyEquals(result[VOTE_ORDERING.targetKey], currentUser.publicKey));
+      assert.equal(result[VOTE_ORDERING.action], VOTE_ACTION.ADD);
+      assert.ok(keyEquals(result[VOTE_ORDERING.voterSigningKey], currentUser.publicSigningKey));
+      assert.equal(result[VOTE_ORDERING.previousChainSignature], undefined);
+      assert.equal(Object.keys(result).length, 7);
+    });
+  });
+});
+
+function keyEquals(arr: number[], uint8array: Uint8Array): boolean {
+  if (Object.keys(arr).length !== uint8array.length) {
```

Comment:
Object.keys on an array? Porque?

Reply:
Hussein had been playing around with it and the arr doesn't have the length property for some reason when we parse it into the JSON.

Reply:
If it for realsies doesn't have a length property, it might not be an array. Screenshot this to me if you get stuck ;)

Reply:
Hussein will shoot you a screenshot soon. What should be happening is a Uint8Array is assigned to a field in a JSON and that JSON is transformed into a Uint8Array. To unpack that data for test, we then convert the Uint8Array to JSON and pull that value out. It has the same data as a Uint8Array, however it is lacking key properties, such as length and other array-like properties.